tests/integration/test_keeper_log_gap_before_committed/test.py

```diff
@@ -0,0 +1,164 @@
+            .split("\n")
+        )
+
+        logging.info(f"Log files: {log_files}")
+        logging.info(f"Snapshot files: {snapshot_files}")
+
+        # We should have multiple changelog files due to max_log_file_size=1
+        assert len(log_files) > 1, f"Expected multiple log files but got: {log_files}"
+        assert len(snapshot_files) >= 1, f"Expected at least one snapshot but got: {snapshot_files}"
+
+        # Get the snapshot to find its index
+        # Snapshot filename format: snapshot_<index>.bin
+        snapshot_file = [f for f in snapshot_files if f.startswith("snapshot_")][0]
+        snapshot_index = int(snapshot_file.split("_")[1].split(".")[0])
+        logging.info(f"Snapshot index: {snapshot_index}")
+
+        def get_from_to_index(changelog):
+            parts = changelog.replace("changelog_", "").replace(".bin", "").split("_")
+            return (int(parts[0]), int(parts[1]))
+
+        # Find a changelog file that is before the snapshot index
+        # Changelog filename format: changelog_<from>_<to>.bin
+        changelog_to_delete = None
+        for log_file in log_files:
+            if log_file.startswith("changelog_"):
+                from_index, to_index = get_from_to_index(log_file)
+
+                # Delete a changelog that is completely before the snapshot
+                # This simulates a gap in committed logs
+                if to_index < snapshot_index and (snapshot_index - to_index) > 4 and from_index > 5:
+                    changelog_to_delete = log_file
+                    break
+
+        assert (
+            changelog_to_delete is not None
+        ), f"Could not find a changelog before snapshot index {snapshot_index}"
+
+        logging.info(f"Deleting changelog: {changelog_to_delete}")
+        node1.exec_in_container(
+            ["rm", f"/var/lib/clickhouse/coordination/log/{changelog_to_delete}"]
+        )
+
+        # Verify the file was deleted
+        log_files_after_delete = (
+            node1.exec_in_container(["ls", "/var/lib/clickhouse/coordination/log"])
+            .strip()
+            .split("\n")
+        )
+        assert (
+            changelog_to_delete not in log_files_after_delete
+        ), f"Changelog {changelog_to_delete} should have been deleted"
+
+        logging.info(f"Log files after deletion: {log_files_after_delete}")
+
+        start_clickhouse()
+        keeper_utils.wait_until_connected(cluster, node1)
+
+        # Reconnect and verify data is still intact
+        node1_conn = get_fake_zk()
+
+        children = node1_conn.get_children("/test_log_gap")
+        assert len(children) == 20, f"Expected 20 children but got {len(children)}"
+
+        for child in children:
+            data = node1_conn.get(f"/test_log_gap/{child}")[0]
+            assert data == b"somedata", f"Data mismatch for {child}: {data}"
+
+        # Verify we can still write new data
+        node1_conn.create("/test_log_gap_after_recovery", b"newdata")
```

Comment:
Does it make sense to have one more restart and check that data is present? This way we can verify that keeper still can figure out the state after "logs before snapshots" were removed.